Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEFileAlgo.cc

```diff
@@ -0,0 +1,401 @@
+                                  << " dy " << dy << " Shift " << xyoff.first << ":" << xyoff.second << " WaferSize "
+                                  << (waferSize + waferSepar);
+#endif
+    for (int u = -N; u <= N; ++u) {
+      for (int v = -N; v <= N; ++v) {
+#ifdef EDM_ML_DEBUG
+        int iu = std::abs(u);
+        int iv = std::abs(v);
+#endif
+        int nr = 2 * v;
+        int nc = -2 * u + v;
+        double xpos = xyoff.first + nc * r;
+        double ypos = xyoff.second + nr * dy;
+        const auto& corner = HGCalGeomTools::waferCorner(xpos, ypos, r, R, rin, rout, false);
+#ifdef EDM_ML_DEBUG
+        ++ntot;
+        if (((corner.first <= 0) && std::abs(u) < 5 && std::abs(v) < 5) || (std::abs(u) < 2 && std::abs(v) < 2)) {
+          edm::LogVerbatim("HGCalGeom") << "DDHGCalEEFileAlgo: " << glog.name() << " R " << rin << ":" << rout
+                                        << "\n Z " << zpos << " LayerType " << layertype << " u " << u << " v " << v
+                                        << " with " << corner.first << " corners";
+        }
+#endif
+        int type = HGCalWaferType::getType(HGCalWaferIndex::waferIndex(layer, u, v, false), waferIndex, waferTypes);
+        if (corner.first > 0 && type >= 0) {
+          int copy = HGCalTypes::packTypeUV(type, u, v);
+#ifdef EDM_ML_DEBUG
+          if (iu > ium)
+            ium = iu;
+          if (iv > ivm)
+            ivm = iv;
+          kount++;
+          if (copies.count(copy) == 0)
+            copies.insert(copy);
+#endif
+          if (corner.first == (int)(HGCalParameters::k_CornerSize)) {
+#ifdef EDM_ML_DEBUG
+            if (iu > iumAll)
+              iumAll = iu;
+            if (iv > ivmAll)
+              ivmAll = iv;
+            ++nin;
+#endif
+            dd4hep::Position tran(xpos, ypos, 0.0);
+            if (layertype > 1)
+              type += 3;
+            glog.placeVolume(ns.volume(wafers[type]), copy, tran);
+#ifdef EDM_ML_DEBUG
+            ++ntype[type];
+            edm::LogVerbatim("HGCalGeom")
+                << " DDHGCalEEFileAlgo: " << wafers[type] << " number " << copy << " type " << layertype << ":" << type
+                << " positioned in " << glog.name() << " at " << tran << " with no rotation";
+#endif
+          }
+        }
+      }
+    }
+
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalEEFileAlgo: Maximum # of u " << ium << ":" << iumAll << " # of v " << ivm
+                                  << ":" << ivmAll << " and " << nin << ":" << kount << ":" << ntot << " wafers ("
+                                  << ntype[0] << ":" << ntype[1] << ":" << ntype[2] << ":" << ntype[3] << ":"
+                                  << ntype[4] << ":" << ntype[5] << ") for " << glog.name() << " R " << rin << ":"
+                                  << rout;
+#endif
+  }
+
+  //Required data members to cache the values from XML file
+  HGCalGeomTools geomTools;
+
+  std::vector<std::string> wafers;     // Wafers
```

Comment:
I think these structs (this one and the ones below) are complex enough that we should follow the CMS code rules: member variables should end in underscore

Reply:
Done that